environments/docker/docker.go

```diff
@@ -48,6 +49,7 @@ type docker struct {
 	Binds       map[string]string   `json:"bindings,omitempty"`
 	NetworkMode string              `json:"networkMode,omitempty"`
 	Network     string              `json:"networkName,omitempty"`
+	AutoRemove  *bool               `json:"autoRemove,omitempty"`
```

Comment:
This is a breaking change. The default of "nil" is going to be false. which means we don't remove the container. We remove them by design, and this changes that behavior entirely, and is a breaking one.

Reply:
Actually, using a pointer reference for this doesn't achieve anything, While the "default" is set in the factory (and you made that true), Golang won't stomp over it if the JSON doesn't have the value and omitempty is used.

Reply:
> This is a breaking change. The default of "nil" is going to be false. which means we don't remove the container. We remove them by design, and this changes that behavior entirely, and is a breaking one.

True, and I actually encountered that bug (server stopped, wouldn't go back up because the container name is hardcoded to the server id), after submittingthat PR :-|